src/main/java/org/opentripplanner/gtfs/mapping/TransferMapper.java

```diff
@@ -253,27 +258,39 @@ private TransferPoint mapTransferPoint(
     throw new IllegalStateException("Should not get here!");
   }
 
-  private int stopPosition(Trip trip, RegularStop stop, Station station, boolean boardTrip) {
+  private int boardStopPosition(Trip trip, RegularStop stop, Station station) {
     List<StopTime> stopTimes = stopTimesByTrip.get(trip);
 
-    // We can board at the first stop, but not alight.
-    final int firstStopPos = boardTrip ? 0 : 1;
-    // We can alight at the last stop, but not board, the lastStopPos is exclusive
-    final int lastStopPos = stopTimes.size() - (boardTrip ? 1 : 0);
-
-    Predicate<StopLocation> stopMatches = station != null
-      ? s -> (s instanceof RegularStop && ((RegularStop) s).getParentStation() == station)
-      : s -> s == stop;
-
-    for (int i = firstStopPos; i < lastStopPos; i++) {
+    for (int i = 0; i < stopTimes.size() - 1; i++) {
       StopTime stopTime = stopTimes.get(i);
-      if (boardTrip && stopTime.getPickupType().isNotRoutable()) {
+      if (stopTime.getPickupType().isNotRoutable()) {
         continue;
       }
-      if (!boardTrip && stopTime.getDropOffType().isNotRoutable()) {
+
+      Predicate<StopLocation> stopMatches = station != null
+        ? s -> ((s instanceof RegularStop regStop) && regStop.getParentStation() == station)
+        : s -> s == stop;
+
```

Comment:
Is there a reason for moving the lambda creation into the loop. Now a new lambda is created for each iteration.